### DISCUSSION TITLE
Strategy for single object keypoint detection without bounding boxes during inference

### DISCUSSION THREAD

**Discussion by NCTU-VRDL** [open-mmlab/mmpose#852] [General]
Hi!

Thanks for the great tools for keypoint detection. I have a custom dataset, and each image contains only a **single object** with 19 key points. I have successfully trained the top-down model using this [config](https://github.com/open-mmlab/mmpose/blob/master/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w48_coco_256x192.py). 

However, in my case, I don't have the ground truth boxes for inference on the new image. For training, I just used the enclosed box that covers all points.

Since I don't want to train another object detection model, should I use the Bottom-up method instead? Any suggestions for this kind of dataset are very welcome. Thanks

**Comment by ly015** [comment]
Yes, a top-down pose model like the one you used would need object bounding boxes during inference. If a detection model is undesired in your case, you could try the bottom-up method, like [associative embedding](https://github.com/open-mmlab/mmpose/tree/master/configs/body/2d_kpt_sview_rgb_img/associative_embedding).

**Reply by zhrRunner** [reply]
> Yes, a top-down pose model like the one you used would need object bounding boxes during inference. If a detection model is undesired in your case, you could try the bottom-up method, like [associative embedding](https://github.com/open-mmlab/mmpose/tree/master/configs/body/2d_kpt_sview_rgb_img/associative_embedding).

hello，I had a problem using your configuration file, KeyError: 'AssociativeEmbedding is not in the mmpose::model registry. Please check whether the value of `AssociativeEmbedding` is correct or it was registered as expected. More details can be found at https://mmengine.readthedocs.io/en/latest/advanced_tutorials/config.html#import-the-custom-module' 
I would be grateful if you could help me

**Comment by jin-s13** [comment]
If the object is roughly at the center of your image, you may choose to train a model that directly takes the whole image as the input.

**Comment by NCTU-VRDL** [comment]
Hi @ly015 
Thanks for the AE method. Do you have any suggestions for the configuration? It looks like I can set the `max_num_people=1`  in `test_cfg`. Anything else?

Hi @jin-s13 
Thanks a lot for your response. Unfortunately, the location of the object varies a lot in each image. I will try to use the whole image as the box instead of the enclosed one.
